File: C#/PriorityQueueApp/PriorityQueue.cs
using System;
using System.Collections.Generic;

namespace PriorityQueueApp
{
    public class PriorityQueue<T>
    {
        private List<(T item, int priority)> elements = new List<(T item, int priority)>();

        public void Enqueue(T item, int priority)
        {
            elements.Add((item, priority));
            elements.Sort((x, y) => x.priority.CompareTo(y.priority));
        }

        public T Dequeue()
        {
            if (elements.Count == 0)
                throw new InvalidOperationException("The queue is empty.");

            var highestPriorityElement = elements[0];
            elements.RemoveAt(0);
            return highestPriorityElement.item;
        }

        public T Peek()
        {
            if (elements.Count == 0)
                throw new InvalidOperationException("The queue is empty.");

            return elements[0].item;
        }
    }
}

File: C/PriorityQueueApp/priority_queue.c
#include <stdio.h>
#include <stdlib.h>
#include "priority_queue.h"

PriorityQueue* create_priority_queue(int capacity) {
    PriorityQueue* pq = (PriorityQueue*)malloc(sizeof(PriorityQueue));
    pq->nodes = (PQNode*)malloc(sizeof(PQNode) * capacity);
    pq->size = 0;
    pq->capacity = capacity;
    return pq;
}

void enqueue(PriorityQueue* pq, void* item, int priority) {
    if (pq->size == pq->capacity) {
        pq->capacity *= 2;
        pq->nodes = (PQNode*)realloc(pq->nodes, sizeof(PQNode) * pq->capacity);
    }
    PQNode node;
    node.item = item;
    node.priority = priority;
    pq->nodes[pq->size++] = node;

    for (int i = pq->size - 1; i > 0 && pq->nodes[i].priority < pq->nodes[i - 1].priority; i--) {
        PQNode temp = pq->nodes[i];
        pq->nodes[i] = pq->nodes[i - 1];
        pq->nodes[i - 1] = temp;
    }
}

void* dequeue(PriorityQueue* pq) {
    if (pq->size == 0) {
        return NULL;
    }
    void* item = pq->nodes[0].item;
    for (int i = 1; i < pq->size; i++) {
        pq->nodes[i - 1] = pq->nodes[i];
    }
    pq->size--;
    return item;
}

void* peek(PriorityQueue* pq) {
    if (pq->size == 0) {
        return NULL;
    }
    return pq->nodes[0].item;
}

void free_priority_queue(PriorityQueue* pq) {
    free(pq->nodes);
    free(pq);
}